runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2914,6 +2918,9 @@ else if (this.authField.equals(MqttConnectProperty.PASSWORD))
                     if (credentialsMatch != null)
                     {
                         sessionAuth = guard.reauthorize(initialId, credentialsMatch);
+                        event.authorization(
+                            sessionAuth == 0 ? Result.FAILURE : Result.SUCCESS,
```

Comment:
Pass the `sessionId` (sessionAuth) to `event.authorization(...)` and handle the mapping to `Result` internally.